src/hasher_impl.rs

```diff
@@ -124,6 +124,57 @@ pub mod blake2s {
     pub type Blake2s256 = Blake2sHasher<U32>;
 }
 
+#[cfg(feature = "blake3")]
+pub mod blake3 {
+    use super::*;
+    use core::marker::PhantomData;
+    use generic_array::typenum::U64;
+
+    // derive_hasher_blake!(blake3, Blake3Hasher, Blake3Digest);
+    derive_digest!(Blake3Digest);
+
+    /// Multihash hasher.
+    #[derive(Debug)]
+    pub struct Blake3Hasher<S: Size> {
+        _marker: PhantomData<S>,
+        hasher: blake3_::Hasher,
+    }
+
+    impl<S: Size> Default for Blake3Hasher<S> {
+        fn default() -> Self {
+            let mut hasher = blake3_::Hasher::new();
+            // TODO
+            // hasher.hash_length(S::to_usize());
+            // Self {
+            //     _marker: PhantomData,
+            //     state: hasher.to_state(),
+            // }
+        }
+    }
+
+    impl<S: Size> StatefulHasher for Blake3Hasher<S> {
+        type Size = S;
+        type Digest = Blake3Digest<Self::Size>;
+
+        fn update(&mut self, input: &[u8]) {
+            self.hasher.update(input);
+        }
+
+        fn finalize(&self) -> Self::Digest {
+            // TODO discuss what to do with xof
+            let digest = self.hasher.finalize();
```

Comment:
Xof?

Reply:
[eXtendable-Output Function](https://www.cryptosys.net/manapi/api_xof.htm). Blake3 doesn't have to have a fixed size output.

Reply:
Ah that's fine. Strobe or blake2b also support variable output sizes. If your question is what sizes to add to the multihash code table, for most applications a 256 or 512 bit hash is what's needed depending if you care about post quantum security or not.